core/src/main/java/org/apache/carbondata/core/readcommitter/LatestFilesReadCommittedScope.java

```diff
@@ -66,7 +68,23 @@ public LatestFilesReadCommittedScope(String path, String segmentId) {
    * @param path carbon file path
    */
   public LatestFilesReadCommittedScope(String path) {
-    this(path, null);
+    this(path, (String) null);
+  }
+
+  /**
+   * a new constructor with path
+   *
+   * @param path carbon file path
+   */
+  public LatestFilesReadCommittedScope(String path, String[] subFolders) {
+    Objects.requireNonNull(path);
+    this.carbonFilePath = path;
+    this.subFolders = subFolders;
```

Comment:
suggest change `subFolders` to `dataFolders` and add javadoc to explain it contains carbondata files and carbonindex files in these folders

Reply:
ok